src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1309,7 +1309,7 @@ public function registerForAutoconfiguration(string $interface)
     /**
      * Registers an attribute that will be used for autoconfiguring annotated classes.
      *
-     * The configurator will receive a ChildDefinition instance, an instance of the attribute and the corresponding \ReflectionClass, in that order.
+     * The configurator will receive a ChildDefinition instance, an instance of the attribute and the corresponding \ReflectionClass or \ReflectionMethod, in that order.
```

Comment:
This is a BC break, as a callback typehinting `ReflectionClass` will break when upgrading to 5.4. We cannot widen the parameter we pass to a callable.

Reply:
We could add `ReflectionMethod` as 4rd nullable parameter?

Reply:
I pushed a change where a 4rd `ReflectionMethod` parameter is added.

Reply:
that could still be a BC break: the callable that is not aware of ReflectionMethod would be called for method-level attributes without handling them properly (as the attribute is not on the ReflectionClass)

Reply:
https://github.com/symfony/symfony/pull/39897#discussion_r561148511 is where I suggested passing the reflection object to the configurator. And as you'll see there, I was already anticipating that we could pass any Reflector there, not only a ReflectionClass.

I think we can and should pass the ReflectionMethod directly. In terms of BC, we should first update the comment in 5.3 to make the intent clear, and then, we could skip caring: any TypeErrror that might result from a too restrictive type-hint will be caught at compile-time. Aka it's trivial to stop. If we want a smoother path, we could check via reflection that passing a ReflectionMethod would be fine and skip if not. But I'm not sure this is worth the extra code and complexity.

Reply:
@nicolas-grekas Shall I partially revert https://github.com/symfony/symfony/pull/42039/commits/6a917af5b92c59d6710806ff6413106a5ab7f390 ?